consensus/src/consensus/consensus_proxy.rs

```diff
@@ -235,4 +246,57 @@ impl<N: Network> ConsensusProxy<N> {
 
         Ok(transactions)
     }
+
+    pub async fn subscribe_transaction_addresses(
+        &self,
+        addresses: Vec<Address>,
+        min_peers: usize,
+        _max: Option<u16>,
+    ) {
+        // First we tell the network to provide us with a vector that contains all the connected peers that support such services
+        // Note: If the network could not provide enough peers that satisfies our requirement, then an error would be returned
+        let peers = self
+            .network
+            .get_peers_by_services(Services::MEMPOOL, min_peers)
+            .await
+            .map_err(|error| {
+                log::error!(
+                    err = %error,
+                    "The transactions by address request couldn't be fulfilled"
+                );
+
+                RequestError::OutboundRequest(OutboundRequestError::SendError)
+            });
+        // Subscribe to any peer that could provide the necessary services
+        for peer_id in peers.unwrap() {
+            let response = self
+                .network
+                .request::<RequestSubscribeToAddress>(
+                    RequestSubscribeToAddress {
+                        operation: TxAddressSubscriptionOperation::Initiate,
+                        addresses: addresses.clone(),
+                    },
+                    peer_id,
+                )
+                .await;
+            match response {
+                Ok(response) => match response.result {
+                    Ok(_) => {
+                        //Done, we are subscribed to a peer
+                        return;
+                    }
+                    Err(subscription_error) => match subscription_error {
+                        crate::SubscribebyAdressErrors::TooManyPeers => todo!(),
+                        crate::SubscribebyAdressErrors::TooManyAddresses => todo!(),
+                        crate::SubscribebyAdressErrors::InvalidOperation => todo!(),
+                        crate::SubscribebyAdressErrors::Other => todo!(),
+                    },
+                },
+                Err(_) => {
+                    //Try with the next peer
+                    continue;
+                }
+            }
+        }
+    }
```

Comment:
This method should return a result that signals if the subscription to at least one peer worked.